File: kinotic-management-api/src/main/java/org/kinotic/management/api/model/UiDeploymentStatus.java
package org.kinotic.management.api.model;

/**
 * Lifecycle state of a {@link UiDeployment}, and why when something is wrong.
 *
 * @param type    the lifecycle state of the deployment
 * @param message why the deployment is in this state, or null when the state speaks for
 *                itself; typically the failure that left it {@link UiDeploymentStatusType#FAILED}
 */
public record UiDeploymentStatus(UiDeploymentStatusType type, String message) {

    public UiDeploymentStatus(UiDeploymentStatusType type) {
        this(type, null);
    }
}

File: kinotic-js/workspace/packages/management-api/src/api/model/UiDeploymentStatus.ts
import { UiDeploymentStatusType } from '@/api/model/UiDeploymentStatusType'

/**
 * Lifecycle state of a UiDeployment, and why when something is wrong.
 */
export interface UiDeploymentStatus {
    /**
     * The lifecycle state of the deployment.
     */
    type: UiDeploymentStatusType
    /**
     * Why the deployment is in this state, or null when the state speaks for itself;
     * typically the failure that left it FAILED.
     */
    message: string | null
}
